src/services/inplayer.account.service.ts

```diff
@@ -44,36 +80,170 @@ export const logout = async () => {
   } catch {
     throw new Error('Failed to fetch user data.');
   }
 };
 
 export const getFreshJwtToken = async ({ auth }: { auth: AuthData }) => auth;
 
-// responsible to convert the InPlayer object to be compatible to the store
-function processInplayerAccount(account: AccountData): Customer {
+export const updateCustomer: UpdateCustomer = async (values) => {
+  try {
+    const fullName = `${values.firstName} ${values.lastName}`;
+
+    const response = await InPlayer.Account.updateAccount({
+      fullName,
+      metadata: {
+        firstName: values.firstName as string,
+        lastName: values.lastName as string,
+        ...(values?.consents && { consents: JSON.stringify(values.consents) }),
+      },
+    });
+
+    return {
+      errors: [],
+      // @ts-ignore
+      // wrong data type from InPlayer SDK (will be updated in the SDK)
+      responseData: processAccount(response.data),
+    };
+  } catch {
+    throw new Error('Failed to update user data.');
+  }
+};
+
+export const getPublisherConsents: GetPublisherConsents = async (config) => {
+  try {
+    const { inplayer } = config.integrations;
+    const { data } = await InPlayer.Account.getRegisterFields(inplayer?.clientId || '');
+
+    // @ts-ignore
+    // wrong data type from InPlayer SDK (will be updated in the SDK)
+    const result: Array<Consent> = data?.collection
+      .filter((field: GetRegisterField) => field.type === 'checkbox')
+      .map((consent: GetRegisterField) => processPublisherConsents(consent));
+
+    return {
+      errors: [],
+      responseData: {
+        consents: [getTermsConsent(), ...result],
+      },
+    };
+  } catch {
+    throw new Error('Failed to fetch publisher consents.');
+  }
+};
+
+export const getCustomerConsents = async (payload: GetCustomerConsentsPayload) => {
+  try {
+    const { customer } = payload;
+
+    if (!customer?.metadata) {
+      return {
+        errors: [],
+        responseData: {
+          consents: [],
+        },
+      };
+    }
+
+    const consents = JSON.parse((customer.metadata?.consents as string) || '');
+
+    return {
+      errors: [],
+      responseData: {
+        consents,
+      },
+    };
+  } catch {
+    throw new Error('Unable to fetch Customer consents.');
+  }
+};
+
+export const updateCustomerConsents: UpdateCustomerConsents = async (payload) => {
+  try {
+    const { customer, consents } = payload;
+
+    const data = {
+      consents,
+      firstName: customer.metadata?.firstName as string,
+      lastName: customer.metadata?.lastName as string,
+    };
+
+    return (await updateCustomer(data, true, '')) as ServiceResponse<never>;
+  } catch {
+    throw new Error('Unable to update Customer`s consents');
+  }
+};
+
+export const getCaptureStatus: GetCaptureStatus = async ({ customer }) => {
+  return {
+    errors: [],
+    responseData: {
+      isCaptureEnabled: true,
+      shouldCaptureBeDisplayed: true,
+      settings: [
+        {
+          answer: {
```

Comment:
It would be nice to add the other register questions that aren't treated as consents here

Reply:
Hmm.. Do we care about `Do you like this form`? I don't see any other question than that.. Or you are talking about something else?

Reply:
`Do you like this form` was just silly test data.  I do think we want to maintain the functionality for companies to add custom questions to the registration flow.

Reply:
Well that's okay but for that to happen we would want to create separate endpoints as cleeng. I don't think this is something important for the MVP. The hardcoded captures and saving in metadata should be okay for first and last name at the moment, but other than that, we shouldn't make it messier.. Maybe leave it for some next phase where we can define better and create separate endpoints.